### PR TITLE
Redirect single person search result to activity

### PR DESCRIPTION
**Pull request by OliHalu**
## Issue

This fixes the following issue(s): 558 -Single results in PersonSearch should proceed to PersonDetails
Now if there's only one search result it will directly proceed to PersonDetails

## Why this is useful for all students
Because its more comy

### CODE REVIEW

File: .idea/codeStyleSettings.xml [hunk 1]
```diff
@@ -1,264 +0,0 @@
-<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kordianbruck) [inline .idea/codeStyleSettings.xml]: Why is this file deleted?

File: .idea/fileTemplates/includes/File Header.java [hunk 2]
```diff
@@ -1,3 +0,0 @@
-/**
```
Comment (kordianbruck) [inline .idea/fileTemplates/includes/File Header.java]: Please don't delete those files

File: .idea/inspectionProfiles/Project_Default.xml [hunk 3]
```diff
@@ -1,115 +0,0 @@
-<component name="InspectionProjectProfileManager">
```
Comment (kordianbruck) [inline .idea/inspectionProfiles/Project_Default.xml]: Please don't delete those files

File: .idea/inspectionProfiles/profiles_settings.xml [hunk 4]
```diff
@@ -1,7 +0,0 @@
-<component name="InspectionProjectProfileManager">
```
Comment (kordianbruck) [inline .idea/inspectionProfiles/profiles_settings.xml]: Please don't delete those files

File: .idea/vcs.xml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <project version="4">
   <component name="VcsDirectoryMappings">
-    <mapping directory="$PROJECT_DIR$" vcs="Git" />
+    <mapping directory="" vcs="Git" />
```
Comment (kordianbruck) [inline .idea/vcs.xml]: ????

File: app/src/main/java/de/tum/in/tumcampusapp/activities/ChatRoomsActivity.java [hunk 6]
```diff
@@ -43,7 +44,7 @@
  * This activity presents the chat rooms of user's
  * lectures using the TUMOnline web service
  */
-public class ChatRoomsActivity extends ActivityForLoadingInBackground<Void, Cursor> implements OnItemClickListener {
+public class ChatRoomsActivity extends ActivityForLoadingInBackground<Void, Cursor>  implements OnItemClickListener {
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/ChatRoomsActivity.java]: Why are you adding a space here?

File: app/src/main/java/de/tum/in/tumcampusapp/activities/ChatRoomsActivity.java [hunk 7]
```diff
@@ -63,6 +64,8 @@ public ChatRoomsActivity() {
     @Override
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        //create BaseActivity for Drawer/up button
+
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/ChatRoomsActivity.java]: What is this?

File: app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java [hunk 8]
```diff
@@ -130,8 +129,21 @@ public void onStartSearch(String query) {
      */
     @Override
     public void onLoadFinished(PersonList response) {
-        if (response.getPersons() == null) {
+        //String "no search result" not shown
+        //change response.getPersons() == null to response.getPersons().isEmpty() or response.getPersons().size() == 0
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java]: This "comment" makes no sense?

File: app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java [hunk 9]
```diff
@@ -130,8 +129,21 @@ public void onStartSearch(String query) {
      */
     @Override
     public void onLoadFinished(PersonList response) {
-        if (response.getPersons() == null) {
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java]: You still need to catch null values. Otherwise it might through a NullPointerException

File: app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java [hunk 10]
```diff
@@ -130,8 +129,21 @@ public void onStartSearch(String query) {
...
+        //String "no search result" not shown
+        //change response.getPersons() == null to response.getPersons().isEmpty() or response.getPersons().size() == 0
+        if (response.getPersons().isEmpty()) {
             lvPersons.setAdapter(new NoResultsAdapter(this));
+        } else if (response.getPersons().size() == 1){
+            ListAdapter adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, android.R.id.text1, response.getPersons());
+            lvPersons.setAdapter(adapter);
+            Person person = (Person) lvPersons.getItemAtPosition(0);
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java]: Why are you adding the item to the layout and then directly moving to another activity? Couldn't you just directly open the new activity, without this detour?

File: app/src/main/java/de/tum/in/tumcampusapp/tumonline/TUMOnlineRequest.java [hunk 11]
```diff
@@ -105,7 +105,7 @@ public void cancelRequest(boolean mayInterruptIfRunning) {
             backgroundTask.cancel(mayInterruptIfRunning);
         }
     }
-
+    //added Utils.setSetting(c, Const.TUMO_DISABLED, false);
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/tumonline/TUMOnlineRequest.java]: ????

File: .idea/codeStyleSettings.xml (hunk 1 above)
Comment (OliHalu) [reply .idea/codeStyleSettings.xml]: Actually I haven't deleted anything at all, as far as I know.

File: app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java (hunk 9 above)
Comment (OliHalu) [reply app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java]: Ok, but then I don't get the "no search result" prompt if there is no result.

File: app/src/main/java/de/tum/in/tumcampusapp/tumonline/TUMOnlineRequest.java (hunk 11 above)
Comment (OliHalu) [reply app/src/main/java/de/tum/in/tumcampusapp/tumonline/TUMOnlineRequest.java]: Oh sorry, that was a (bad) try to get rid of the token identification bug to solve one of the chatroom issues

File: .idea/codeStyleSettings.xml (hunk 1 above)
Comment (kordianbruck) [reply .idea/codeStyleSettings.xml]: Clearly the diff says differently :)

File: app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java (hunk 9 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java]: You can do both. `if(null or isEmpty()`

File: app/src/main/java/de/tum/in/tumcampusapp/tumonline/TUMOnlineRequest.java (hunk 11 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/tumonline/TUMOnlineRequest.java]: Ahhh, right. The diff catches all of those forgotten pieces of debugging attempts. 
(Also: theres a Debug button in your IDE for such cases :+1: )

File: app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java [hunk 12]
```diff
@@ -130,8 +129,21 @@ public void onStartSearch(String query) {
...
+            // store selected person ID in bundle to get in in StaffDetails
+            Bundle bundle = new Bundle();
+            bundle.putSerializable("personObject", person);
+            // show detailed information in new activity
+            Intent intent = new Intent(this, PersonsDetailsActivity.class);
+            intent.putExtras(bundle);
+            startActivity(intent);
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/activities/PersonsSearchActivity.java]: This looks like copypasta'd 🍝  code to me.

I'd just reuse the onItemClick method here, but moving the duplicated code to a separate method would probably be better.
